midend/tableHit.h

```diff
@@ -35,13 +35,26 @@ This may be needed by some back-ends which only support hit test in conditionals
 */
 class DoTableHit : public Transform, public ResolutionContext {
     TypeMap *typeMap;
+    enum op_t { None, And, Or, Xor };
+
+    const IR::Node *process(IR::AssignmentStatement *statement, op_t op);
+    const IR::Node *postorder(IR::AssignmentStatement *statement) override {
+        return process(statement, None);
+    }
+    const IR::Node *postorder(IR::OpAssignmentStatement *statement) override { return statement; }
+    const IR::Node *postorder(IR::BAndAssign *statement) override {
+        return process(statement, And);
+    }
+    const IR::Node *postorder(IR::BOrAssign *statement) override { return process(statement, Or); }
+    const IR::Node *postorder(IR::BXorAssign *statement) override {
+        return process(statement, Xor);
+    }
```

Comment:
Can this happen? Currently typechecker requires the type in op-assign to be `Type_Bits`, but here it would need to be `Type_Boolean`. It is the typeching that probably should be relaxed.

Reply:
Typechecking allows the use of `&`, `|`, and `^` on bool, so I had it allow corresponding op-assigns.